### PR TITLE
First branch

### PR DESCRIPTION
**Pull request by dashok-myr**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "next/core-web-vitals"
```
Comment (tp00012x) [inline .eslintrc.json:2]: love you

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+# local env files
+.env*.local
+
+# vercel
+.vercel
+
+# typescript
+*.tsbuildinfo
+next-env.d.ts
```
Comment (tp00012x) [inline .gitignore:35]: add: `.idea` here too, so you don't add Webstorm config folder

File: src/app/globals.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
+
+body {
+  background-color: #1a2a33
```
Comment (tp00012x) [inline src/app/globals.css]: is this correct? We probably want to use maybe one of Tailwind colors, or if Tailwind doesn't have the color that we need, we will need to add it to the tailwind.config

You can't have a random `#1a2a33` in the code. This has to be part of the color design library

File: public/next.svg
Comment (tp00012x) [inline public/next.svg:1]: do we need this?

File: public/vercel.svg
Comment (tp00012x) [inline public/vercel.svg:1]: Do we need this?

File: src/app/globals.css (hunk 3 above)
Comment (tp00012x) [reply src/app/globals.css]: I think this is `dark-green` in tailwind

you can do i believe:

```
body {
  @apply bg-dark-green
}
```

File: src/app/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  title: "Tic Tac Toe",
+  description: "Tic Tac Toe Game",
+};
+
+export default function RootLayout({ children }: { children: ReactNode }) {
+  return (
+    <html lang="en">
+      <body className={inter.className}>
+        <PlayersProvider>
+          <GameScoreProvider>
```
Comment (tp00012x) [inline src/app/layout.tsx]: I don't think we need `GameScoreProvider` to be wrapping the whole app, just the component where we actually right? Like the game selecion mode screen doesn't care about this. We should move this down

File: src/components/NewGameButtons.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import Link from "next/link";
+import React from "react";
+
+export default function NewGameButtons() {
```
Comment (tp00012x) [inline src/components/NewGameButtons.tsx]: We don't need this abstraction:

It should be something like this in `Home.tsx` component:

```
<main className="flex ...">
        <div className="w-full flex flex-col gap-10">
          <Image
            src={logo}
            alt="x"
            width="80"
            height="80"
            className="self-center"
          />
          <ChooseMarkSwitch />
          <div className="flex flex-col gap-6">
            <Button1WithAGoodName />
            <Button2WithAGoodName />
          </div
        </div>
      </main>



```

File: src/components/Modal.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  firstBtnLabel,
+  secondBtnLabel,
+  showModal,
+  navigate,
+  onFirstBtnClick,
+  onSecondBtnClick,
+}: ModalProps) {
+  return (
+    <>
+      {showModal ? (
```
Comment (tp00012x) [inline src/components/Modal.tsx]: Use early return pattern. This look ugly. Avoid nesting. Watch: https://www.youtube.com/watch?v=CFRhGnuXG-4

File: src/components/Modal.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  onFirstBtnClick: () => void;
+  onSecondBtnClick: () => void;
+}
+
+export default function Modal({
+  gameResult,
+  firstBtnLabel,
+  secondBtnLabel,
+  showModal,
+  navigate,
```
Comment (tp00012x) [inline src/components/Modal.tsx]: This name is not a good name. There are 2 buttons here. It's not clear what button navigates

File: src/app/startgame/RestartButton.tsx [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import Image from "next/image";
+import restart from "@/icons/icon-restart.svg";
+import React from "react";
+
+interface IRestartButton {
+  onRestart: () => void;
+}
+
+export default function RestartButton({ onRestart }: IRestartButton) {
```
Comment (tp00012x) [inline src/app/startgame/RestartButton.tsx:9]: awesome, great job

File: src/app/startgame/TurnLogo.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+import Image from "next/image";
+import cross from "@/icons/icon-x.svg";
+import nought from "@/icons/icon-o.svg";
+import React from "react";
+
+interface ITurnLogo {
+  currentPlayerMark: EMark;
+}
+
+export default function TurnLogo({ currentPlayerMark }: ITurnLogo) {
```
Comment (tp00012x) [inline src/app/startgame/TurnLogo.tsx:11]: <3

File: src/app/startgame/page.tsx [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+            setGameState(GameState.RESTART);
+            setPlayers({
+              p1: { mark: EMark.CROSS, slots: [] },
+              p2: { mark: EMark.NOUGHT, slots: [] },
+            });
+          }}
+        />
+      </div>
+      <div className="grid grid-cols-3 gap-4 justify-between">
+        {cellDesign.map((_, index) => {
```
Comment (tp00012x) [inline src/app/startgame/page.tsx]: Let's move this into its own component. So the levels of abstraction is the same here.

Like this component is a higher overview, smart component, no too many details

File: src/app/startgame/page.tsx [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+        ties: gameScore.ties + 1,
+      });
+    }
+    setPlayers(copyPlayers);
+    setCurrentPlayer(currentPlayer === "p1" ? "p2" : "p1");
+  }
+
+  return (
+    <div className="flex flex-col justify-center gap-10 mx-auto w-1/2 md:w-1/4 h-screen">
+      <Modal
```
Comment (tp00012x) [inline src/app/startgame/page.tsx:107]: Let's move this out of this div:


```
<>
  <div className="flex flex-col justify-center gap-10 mx-auto w-1/2 md:w-1/4 h-screen">
  </div>
  <Modal ...
</>
```

The modal is its own thing, let's not pollute this div that actually gets affected but its layout

File: src/components/ChooseMarkSwitch.tsx [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+          <button
+            onClick={() => {
+              setCurrentMark(EMark.NOUGHT);
+              const copyPlayerObj = { ...players };
+              copyPlayerObj.p1.mark = EMark.NOUGHT;
+              copyPlayerObj.p2.mark = EMark.CROSS;
+              setPlayers(copyPlayerObj);
+            }}
+            className={`flex-1 ${
+              currentMark === EMark.NOUGHT ? "p-3 bg-silver rounded-xl" : ""
```
Comment (tp00012x) [inline src/components/ChooseMarkSwitch.tsx]: Should we install `clsx` or `classname`? to handle nicely the conditional class names

File: src/app/startgame/RestartButton.tsx (hunk 8 above)
Comment (dashok-myr) [reply src/app/startgame/RestartButton.tsx:9]: Actually, the icon doesn’t display for some reason

File: src/app/startgame/page.tsx (hunk 10 above)
Comment (dashok-myr) [reply src/app/startgame/page.tsx]: gotta need help here! I tried to do it but the <MarkIcon /> component wasn't working properly

File: src/components/ChooseMarkSwitch.tsx (hunk 12 above)
Comment (dashok-myr) [reply src/components/ChooseMarkSwitch.tsx]: classnames it is. But I don't understand why I should use the external library if the default condition statement also works.

File: src/components/Modal.tsx (hunk 6 above)
Comment (dashok-myr) [reply src/components/Modal.tsx]: you are ugly

**Pull request merged**